### DISCUSSION TITLE
undefined reference to some functions in Crc_8H2C.c and std_timer.c  ||  Crc_8H2C.c和std_timer.c两个文件中有多个函数报引用未定义错误

### DISCUSSION THREAD

**Discussion by WhichOne-Studio** [autoas/ssas-public#33] [Q&A]
C:\ssas-public-master>scons --cpl=QemuVersatilepbGCC --app=VersatilepbCanApp
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
scons: building associated VariantDir targets: build\nt\QemuVersatilepbGCC\VersatilepbCanApp
LINK build\nt\QemuVersatilepbGCC\VersatilepbCanApp\VersatilepbCanApp.exe
build\nt\QemuVersatilepbGCC\VersatilepbCanApp\infras\communication\CanTSyn\CanTSyn.o: In function `CanTSyn_HandleMsgSync':
C:\ssas-public-master/infras\communication\CanTSyn/CanTSyn.c:71: undefined reference to `Crc_CalculateCRC8H2F'
C:\ssas-public-master/infras\communication\CanTSyn/CanTSyn.c:117: undefined reference to `StbM_GetCurrentVirtualLocalTime'
build\nt\QemuVersatilepbGCC\VersatilepbCanApp\infras\communication\CanTSyn\CanTSyn.o: In function `CanTSyn_HandleMsgFup':
C:\ssas-public-master/infras\communication\CanTSyn/CanTSyn.c:176: undefined reference to `Crc_CalculateCRC8H2F'
C:\ssas-public-master/infras\communication\CanTSyn/CanTSyn.c:218: undefined reference to `StbM_GetCurrentVirtualLocalTime'
C:\ssas-public-master/infras\communication\CanTSyn/CanTSyn.c:235: undefined reference to `StbM_BusSetGlobalTime'
build\nt\QemuVersatilepbGCC\VersatilepbCanApp\infras\communication\CanTSyn\CanTSyn.o: In function `CanTSyn_TransmitSYNC':
C:\ssas-public-master/infras\communication\CanTSyn/CanTSyn.c:261: undefined reference to `StbM_GetCurrentVirtualLocalTime'
C:\ssas-public-master/infras\communication\CanTSyn/CanTSyn.c:276: undefined reference to `Crc_CalculateCRC8H2F'
build\nt\QemuVersatilepbGCC\VersatilepbCanApp\infras\communication\CanTSyn\CanTSyn.o: In function `CanTSyn_TransmitFUP':
C:\ssas-public-master/infras\communication\CanTSyn/CanTSyn.c:350: undefined reference to `Crc_CalculateCRC8H2F'
build\nt\QemuVersatilepbGCC\VersatilepbCanApp\infras\communication\CanTSyn\CanTSyn.o: In function `CanTSyn_MainMaster':
C:\ssas-public-master/infras\communication\CanTSyn/CanTSyn.c:390: undefined reference to `StbM_GetTimeBaseUpdateCounter'
build\nt\QemuVersatilepbGCC\VersatilepbCanApp\infras\communication\CanTSyn\CanTSyn.o: In function `CanTSyn_TxConfirmation':
C:\ssas-public-master/infras\communication\CanTSyn/CanTSyn.c:503: undefined reference to `StbM_GetCurrentVirtualLocalTime'
scons: *** [build\nt\QemuVersatilepbGCC\VersatilepbCanApp\VersatilepbCanApp.exe] Error 1
scons: building terminated because of errors.

when i compile the **QEMU Versatilepb bootloader demo**, a couple of functions' definition in **Crc_8H2C.c** and **std_timer.c** cannot be found by the compiler. Can anyone help me? Thanks a lot!
我编译**QEMU Versatilepb bootloader demo**时报错显示**Crc_8H2C.c**和**std_timer.c**两个文件中有多个函数引用未定义，求助！非常感谢！

**Comment by parai** [comment]
this issue has been addressed by commit https://github.com/autoas/ssas-public/commit/b8174bca3741ca071c87c4cb5ffadf82d0775b7f
